code/modules/power/singularity/emitter.dm

```diff
@@ -253,6 +253,7 @@
 		else
 			if(QDELETED(target))
 				target = null
+	P.firer = src

```

Comment:
No.
```firer``` is typed as a mob.

If you're wanting to do this, then remove all assumptions that it's a mob (ie: go through all uses of ```firer```)